### PR TITLE
Break up discovery

### PR DESCRIPTION
**Pull request by tsjohns9**

### CODE REVIEW

File: pkg/discovery/agent.go [hunk 1]
```diff
@@ -18,38 +15,44 @@ import (
...
-	publishBufferSize   int
-	stage               string
-	stopAgent           chan bool
-	stopDiscovery       chan bool
-	stopPublish         chan bool
+	anypointClient anypoint.Client
+	assetCache     cache.Cache
+	stage          string
+	stopAgent      chan bool
+	stopDiscovery  chan bool
```
Comment (sagarkal) [inline pkg/discovery/agent.go:22]: stopDisovery and stopPublish can be removed from the Agent struct then right?

File: pkg/discovery/agent.go [hunk 2]
```diff
@@ -88,14 +86,14 @@ func (a *Agent) Run() {
...
-	go a.publishLoop()
+	go a.discovery.DiscoveryLoop()
+	go a.publisher.PublishLoop()
 
 	select {
 	case <-a.stopAgent:
 		log.Info("Received request to kill agent")
-		a.stopDiscovery <- true
-		a.stopPublish <- true
+		a.discovery.Stop()
```
Comment (sagarkal) [inline pkg/discovery/agent.go:95]: I guess we need to include a scenario for graceful exit as well

File: pkg/discovery/agent.go (hunk 1 above)
Comment (tsjohns9) [reply pkg/discovery/agent.go:22]: yeah I think so since the new structs have a stop method

File: pkg/discovery/agent.go (hunk 2 above)
Comment (tsjohns9) [reply pkg/discovery/agent.go:95]: What does that look like? I thought if the agent is being stopped it is already a graceful exit

**Pull request closed without merging**